pocs/observatory.py

```diff
@@ -537,6 +555,235 @@ def close_dome(self):
+        of the `(max_counts + min_counts) - bias`.
+
+        The next exposure time is calculated as:
+
+        ```
+            exp_time = int(previous_exp_time * (target_adu / counts) *
+                       (2.0 ** (elapsed_time / 180.0)) + 0.5)
+        ```
+
+        Under- and over-exposed images are rejected. If image is saturated with
+        a short exposure the method will wait 60 seconds before beginning next
+        exposure.
+
+        Optionally, the method can also take dark exposures of equal exposure
+        time to each flat-field image.
+
+
+        Args:
+            alt (float, optional): Altitude for flats
+            az (float, optional): Azimuth for flats
+            min_counts (int, optional): Minimum ADU count
+            max_counts (int, optional): Maximum ADU count
+            bias (int, optional): Default bias for the cameras
+            max_exptime (float, optional): Maximum exposure time before stopping
+            camera_list (list, optional): List of cameras to use for flat-fielding
+            target_adu_percentage (float, optional): Exposure time will be adjust so
+                that counts are close to: target * (`min_counts` + `max_counts`). Default
+                to 0.5
+            max_num_exposures (int, optional): Maximum number of flats to take
+            take_darks (bool, optional): If dark fields matching flat fields should be
+                taken, default False.
+            *args (TYPE): Description
+            **kwargs (TYPE): Description
+        """
+        target_adu = target_adu_percentage * (min_counts + max_counts)
+
+        image_dir = self.config['directories']['images']
+
+        flat_obs = self._create_flat_field_observation(alt=alt, az=az)
+
+        if camera_list is None:
+            camera_list = list(self.cameras.keys())
+
+        # Start with 1 second exposures for each camera
+        exp_times = {cam_name: [1. * u.second] for cam_name in camera_list}
+
+        # Loop until conditions are met for flat-fielding
+        while True:
+            self.logger.debug("Slewing to flat-field coords: {}".format(flat_obs.field))
+            self.mount.set_target_coordinates(flat_obs.field)
+            self.mount.slew_to_target()
+            self.status()  # Seems to help with reading coords
+
+            fits_headers = self.get_standard_headers(observation=flat_obs)
+
+            start_time = current_time()
+            fits_headers['start_time'] = utils.flatten_time(start_time)
+
+            camera_events = dict()
+
+            # Take the observations
+            for cam_name in camera_list:
+
+                camera = self.cameras[cam_name]
+                exp_time = exp_times[cam_name][-1].value
+
+                filename = "{}/flats/{}/{}/{}.{}".format(
+                    image_dir,
+                    camera.uid,
+                    flat_obs.seq_time,
+                    'flat_{:02d}'.format(flat_obs.current_exp),
```

Comment:
Why not just put that into the filename format string?

Reply:
Too much copying and pasting, not enough thinking. :)